codegen/src/api/errors.rs

```diff
@@ -74,9 +106,16 @@ pub fn generate_has_module_error_impl(
             }
         }
     } else {
+        let error_conversion = if module_legacy_err {
+            quote! { module_error.error }
+        } else {
+            // Convert [u8; 4] errors to legacy format.
```

Comment:
Error number is obtained as per: https://github.com/paritytech/substrate/pull/10242/files#diff-a8382c7a11352cf83fe1e60f06e0a9059a00443155b0a8e3189de37bc9ac4fedR1586.
Wondering if we could do more here. What do you think?

Reply:
I'm not sure what the other 3 bytes in that new `[u8; 4]` will do yet; it's prolly worth trying to find out (perhaps a question in dumb questions to see who answers?). But offhand, I'm sortof happy to do either: 

- drop back to the legacy single-u8 and accept that information will be lost (whatever it is), or 
- always end up with a `[u8; 4]` so that we preserve whatever additional info is there (I get the impression that setting the other 3 bytes to 0 when porting from the legacy formats will be ok).

@ascjones any thoughts?

Reply:
I think while we are doing this, we should at the minimum expose the raw bytes so the user can extract any information that might be in there now or in the future. 

One way would be to modify [`HasModuleError`:](https://github.com/paritytech/subxt/blob/e866d744de5e8c4d43b5673701dd6b5232a1b7d1/subxt/src/error.rs#L190) to give access to the `[u8; 4]` possibly by changing the return type of `(u8, u8)` to include that (maybe promote to a struct).

Better than throwing that away, and once usage patterns emerge of that `[u8; 4]` we can add some structure to it.

Reply:
That makes sense, have updated it with a `ModuleErrorRaw` that is exposed to the customer on the error path, as part of the `ModuleError`. 
We still do the conversion to error string assuming the error index is indeed the first byte. We could extend it further once the XCM is introduced.
What do you think? 😄

Reply:
Yes looks good I have added some nitpicks see below